ext/msgpack/packer.c

```diff
@@ -124,8 +124,18 @@ void msgpack_packer_write_hash_value(msgpack_packer_t* pk, VALUE v)
 void msgpack_packer_write_other_value(msgpack_packer_t* pk, VALUE v)
 {
     int ext_type;
-    VALUE proc = msgpack_packer_ext_registry_lookup(&pk->ext_registry,
-            rb_obj_class(v), &ext_type);
+
+    VALUE lookup_class;
+
+    if (SYMBOL_P(v)) { /* Symbols have no singleton class */
```

Comment:
I took a look at [`singleton_class_of`](https://github.com/ruby/ruby/blob/54a77d2044148bb442340f93747e1f345b6ca401/class.c#L1587) and the cases for which a TypeError is raised are Integer (Fixnum, Bignum), Float, Symbol and frozen String. Since all but symbols are already filtered out when reaching this code I left out the checks for them.